crates/dojo/core/src/world/world_contract.cairo

```diff
@@ -505,6 +501,8 @@ pub mod world {
                 )
             };
 
+            self.assert_resource_upgradability(event_selector, prev_address, new_contract_address);
```

Comment:
We have the namespace, we should pass a snapshot of the `ByteArray` for the error message in the assert.